### PR TITLE
minor test/pylib type fixes

### PR DESCRIPTION
**Pull request by alecco**
Some minor fixes reported by `mypy`.

### CODE REVIEW

**Comment by alecco** [comment]
> CI state **`FAILURE`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/2474/

restarted

**Comment by alecco** [comment]
Changes in https://github.com/scylladb/scylladb/commit/de5ae8adf9e3c82423c56089b11f9e3a55e25147:

- rebased and fixed conflict

**Comment by avikivity** [comment]
> > CI state **`FAILURE`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/2474/
> 
> restarted

Please file issues for CI failures (or leave a leave a comment, like and subscribe if an issue exists).

**Comment by alecco** [comment]
> > > CI state **`FAILURE`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/2474/
> > 
> > 
> > restarted
> 
> Please file issues for CI failures (or leave a leave a comment, like and subscribe if an issue exists).

In this case it was my fault, I forgot to rebase to the very latest master.

**Comment by tchaikov** [comment]
> > > > CI state **`FAILURE`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/2474/
> > > 
> > > 
> > > restarted
> > 
> > 
> > Please file issues for CI failures (or leave a leave a comment, like and subscribe if an issue exists).
> 
> In this case it was my fault, I forgot to rebase to the very latest master and I missed seeing the merge conflict in the PR.

FWIW, just added a new failure cause rule to detect this kind of failures.

**Review by kostja — APPROVED** [review]

**Review by kbr-scylla — COMMENTED** [review]
One small request.

File: test/pylib/rest_client.py [hunk 1]
```diff
@@ -237,7 +237,7 @@ async def message(self) -> None:
         await self.api.message_injection(self.node_ip, self.injection)
 
 @asynccontextmanager
-async def inject_error(api: ScyllaRESTAPIClient, node_ip: IPAddress, injection: str, parameters: dict[str, Any] = {}) -> InjectionHandler:
+async def inject_error(api: ScyllaRESTAPIClient, node_ip: IPAddress, injection: str, parameters: dict[str, Any] = {}):
```
[the remark this answers is not in the document]
Comment (alecco) [reply test/pylib/rest_client.py]: Fixed in https://github.com/scylladb/scylladb/commit/83c99df1010904dd36abf0411b513b8ff3f9f32d

**Review by kbr-scylla — APPROVED** [review]

**Comment by alecco** [comment]
Changes in https://github.com/scylladb/scylladb/commit/83c99df1010904dd36abf0411b513b8ff3f9f32d:

- fix return type (Kamil)

Changes in https://github.com/scylladb/scylladb/commit/8fceb7b7a0dc14e1259e5a5305d6c89afe778017:

- rebase on latest master

**Pull request merged**